applications/CompressiblePotentialFlowApplication/custom_elements/incompressible_adjoint_potential_flow_element.h

```diff
@@ -0,0 +1,174 @@
+
+    /**
+     * Destructor
+     */
+    ~AdjointIncompressiblePotentialFlowElement() override {};
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /// Assignment operator.
+    AdjointIncompressiblePotentialFlowElement & operator=(AdjointIncompressiblePotentialFlowElement const& rOther)
+    {
+        BaseType::operator=(rOther);
+        Flags::operator =(rOther);
+        return *this;
+    }
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    Element::Pointer Create(IndexType NewId, NodesArrayType const& ThisNodes, PropertiesType::Pointer pProperties) const override;
+
+    Element::Pointer Create(IndexType NewId, GeometryType::Pointer pGeom, PropertiesType::Pointer pProperties) const override;
+
+    Element::Pointer Clone(IndexType NewId, NodesArrayType const& ThisNodes) const override;
+
+    IntegrationMethod GetIntegrationMethod() const override;
+
+    void Initialize() override;
+
+    void InitializeSolutionStep(ProcessInfo& rCurrentProcessInfo) override;
+
+    void FinalizeSolutionStep(ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateLocalSystem(MatrixType& rLeftHandSideMatrix,
+                                      VectorType& rRightHandSideVector,
+                                      ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateLeftHandSide(MatrixType& rLeftHandSideMatrix,
+                                       ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateRightHandSide(VectorType& rRightHandSideVector,
+                                        ProcessInfo& rCurrentProcessInfo) override;
+
+    void GetValueOnIntegrationPoints(const Variable<double>& rVariable,
+            std::vector<double>& rValues,
+            const ProcessInfo& rCurrentProcessInfo) override;
+    
+    void GetValueOnIntegrationPoints(const Variable<array_1d<double,3> >& rVariable,
+            std::vector< array_1d<double,3> >& rValues,
+            const ProcessInfo& rCurrentProcessInfo) override;
+
+    void GetValuesVector(Vector& rValues, int Step=0) override;
+
+    void CalculateSensitivityMatrix(const Variable<array_1d<double,3> >& rDesignVariable,
+                                            Matrix& rOutput,
+                                            const ProcessInfo& rCurrentProcessInfo) override;
+
+    void EquationIdVector(EquationIdVectorType& rResult, ProcessInfo& CurrentProcessInfo) override;
+
+    void GetDofList(DofsVectorType& rElementalDofList, ProcessInfo& CurrentProcessInfo) override;
+
+    int Check(const ProcessInfo& rCurrentProcessInfo) override;
+
+    std::string Info() const override;
+
+    void PrintInfo(std::ostream& rOStream) const override;
+
+    void PrintData(std::ostream& rOStream) const override;
+
+   Element::Pointer pGetPrimalElement();
+
+
+protected:
+
+
+    Element::Pointer mpPrimalElement;
+
+    void GetWakeDistances(array_1d<double,NumNodes>& distances);
```

Comment:
Is there a special reason why these functions are protected and not private?

Reply:
If they are protected they can be called (at some point) by child classes. So for instance if we need to define a compressible_adjoint, or embedded_adjoint, they would inherit these members and functions from this class.